src/h5web/dataset-visualizer/vis/MatrixVis.tsx

```diff
@@ -13,34 +21,179 @@ interface Props {
   data: HDF5Value;
 }
 
+type CellProps = Omit<GridChildComponentProps, 'data'>;
+
+interface MatrixProps {
+  children: CellRendererFunc;
+  width: number;
+  height: number;
+}
+
+interface IndexCellProps {
+  index: number;
+  style: React.CSSProperties;
+}
+
+type CellRendererFunc = (props: GridChildComponentProps) => JSX.Element;
+
 function MatrixVis(props: Props): JSX.Element {
   const { dims, data } = props;
+  const [sticky, setSticky] = useState(true);
+  const gridDimensions = {
+    rowHeight: 32,
+    rowCount: dims[0] + 1,
+    columnWidth: 116,
+    columnCount: (dims.length === 2 ? dims[1] : 1) + 1,
+  };
+  const GridSettings = createContext(gridDimensions);
+
+  const CellRenderer: CellRendererFunc = (
```

Comment:
I'd prefer to keep the `function` declaration syntax for readability and consistency: `function CellRenderer(...): ... { ... }` even if it means you can't reuse type `CellRendererFunc` (you already type the argument anyway).